src/test/ProposalPayloadE2E.t.sol

```diff
@@ -3,37 +3,335 @@ pragma solidity ^0.8.15;
+        //     liquidationThreshold: 7000,
+        //     liquidationBonus: 10800,
+        //     liquidationProtocolFee: 1000,
+        //     reserveFactor: 2000,
+        //     usageAsCollateralEnabled: true,
+        //     borrowingEnabled: false,
+        //     interestRateStrategy: _findReserveConfigBySymbol(allConfigsAfterV3Polygon, "BAL")
+        //         .interestRateStrategy,
+        //     stableBorrowRateEnabled: false,
+        //     isActive: true,
+        //     isFrozen: false,
+        //     isSiloed: false,
+        //     isBorrowableInIsolation: false,
+        //     isFlashloanable: false,
+        //     supplyCap: 100_000_000,
+        //     borrowCap: 256_140,
+        //     debtCeiling: 2_000_000_00,
+        //     eModeCategory: 1
+        // });
+
+        // InterestStrategyValues memory expectedStrategy = _getExpectedStrategyPolygon();
+
+        // _validateReserveConfig(expectedAssetConfig, allConfigsAfterV3Polygon);
+        // _validateInterestRateStrategy(
+        //     ProposalPayloadPolygon(proposalPayloadPolygon).INTEREST_RATE_STRATEGY(),
+        //     ProposalPayloadPolygon(proposalPayloadPolygon).INTEREST_RATE_STRATEGY(),
+        //     expectedStrategy
+        // );
+    }
+
+    function testUtilizationAtZeroPercent() public {
+        (uint256 liqRate, uint256 stableRate, uint256 varRate) = strategy.calculateInterestRates(
+            BAL,
+            100 * 1e18,
+            0,
+            0,
+            68200000000000000000000000,
+            2000
+        );
+
+        // At nothing borrowed, liquidity rate should be 0, variable rate should be 3% and stable rate should be 3%.
+        assertEq(liqRate, 0);
+        assertEq(stableRate, 3 * (AaveV2Helpers.RAY / 100));
+        assertEq(varRate, 3 * (AaveV2Helpers.RAY / 100));
+    }
+
+    function testUtilizationAtOneHundredPercent() public {
+        (uint256 liqRate, uint256 stableRate, uint256 varRate) = strategy.calculateInterestRates(
+            BAL,
+            0,
+            0,
+            100 * 1e18,
+            68200000000000000000000000,
+            2000
+        );
+
+        // At max borrow rate, stable rate should be 87% and variable rate should be 85.75%.
+        assertEq(liqRate, 1336000000000000000000000000);
+        assertEq(stableRate, 3130000000000000000000000000);
+        assertEq(varRate, 1670000000000000000000000000);
+    }
+
+    function testUtilizationAtUOptimal() public {
+        (uint256 liqRate, uint256 stableRate, uint256 varRate) = strategy.calculateInterestRates(
+            BAL,
+            20 * 1e18,
+            0,
+            80 * 1e18,
+            68200000000000000000000000,
+            2000
+        );
+
+        // At UOptimal, stable rate should be % and variable rate should be 14%.
+        assertEq(liqRate, 108800000000000000000000000);
+        assertEq(stableRate, 13 * (AaveV2Helpers.RAY / 100));
+        assertEq(varRate, 17 * (AaveV2Helpers.RAY / 100));
+    }
```

Comment:
Would recommend adding similar tests for the Polygon V2 and V3 strategies as well